tests/aggregation/tests.py

```diff
@@ -395,6 +395,37 @@ def test_count(self):
         vals = Book.objects.aggregate(Count("rating", distinct=True))
         self.assertEqual(vals, {"rating__count": 4})
 
+    def test_non_grouped_annotation_not_in_group_by(self):
+        """
+        An annotation not included in `values()` before an aggregate should be
+        excluded from the group by clause.
+        """
+        qs = (
+            Book.objects.annotate(xprice=F('price')).filter(rating=4.0).values('rating')
+                .annotate(count=Count('publisher_id', distinct=True)).values('count', 'rating').order_by('rating')
+        )
+        self.assertEqual(
+            list(qs), [
+                {'rating': 4.0, 'count': 2},
+            ]
+        )
+
+    def test_grouped_annotation_in_group_by(self):
+        """
+        An annotation included in `values()` before an aggregate should be
+        included in the group by clause.
+        """
+        qs = (
+            Book.objects.annotate(xprice=F('price')).filter(rating=4.0).values('rating', 'xprice')
+                .annotate(count=Count('publisher_id', distinct=True)).values('count', 'rating').order_by('rating')
+        )
+        self.assertEqual(
```

Comment:
@timgraham is ordering by the result of an aggregate allowed without subqueries? If ordering by count does not error, then it should be safe to use that ordering. If not, introducing a different field into the orderby will affect the grouping (not that you suggested that), so we'll need to look at comparing the queryset out of order if there's another assert method available that does that.

I'm not able to check either of these things at the moment, but I can take a look in about 8 hours if it's not resolved.